### ISSUE TITLE
Feature Request: Remember layout depending on which monitors are present

### ISSUE THREAD

**Issue by stevesimmons** [gTile/gTile#55]
Hi,

I've just stumbled across gTile and it sounds like the solution to my longstanding tiling annoyances with stock Ubuntu.

Is it possible to have several different tiling layouts depending on which monitors are present? I use a laptop which is connected to a 4k external monitor at work, a smaller 2560x1440 one at home, and when I am traveling just the laptop screen. It would be great if I could define gTile layouts for each setup, and have gTile use the right one automatically, without me having to manually switch.

Is this possible today, or could it be easily added?

Thanks!

**Comment by scherepanov** [comment]
Yes, this is part of bigger problem. It is not implemented today.
I have at office horizontal monitor and two verticals on sides, connected to same desktop. gTile layout optimized for horizontal layout, on vertical still works and usable but looks odd. In theory, you need to have separate layout for each monitor, on same desktop.
Next, I am using RDP to connect to same desktop. What makes it interesting, RDP window can be resized. You cannot automatically assign monitor layout based on screen size or screen sides ratio, because it is not constant. At same time, RDP can spawn several monitors.
It is possible to implement several sets of presets (only resize presets, keyboard presets should stay same). And switch between presets in preferences. It will cover your use case. Would be nice to assign preset to monitor. This is harder to do as seems to be required some GUI.
Nice feature to have. Will look what is possible to do.
Or, take a fate in your own hands, and submit PR.

**Comment by stevesimmons** [comment]
> Or, take a fate in your own hands, and submit PR.
If only I had any spare time... I have too many conflicting deliverables at the moment unfortunately.

Since I'm typing now, I'll add my other first impression as a new user. gTile could have a second set of accelerators better suited for laptops with no keypad. Then a laptop install would default to Super+Return say instead of Super+Enter, and similarly for the number keys. This remapping isn't hard to do manually, once you realize it's what you need to do, but it would save some confused moments for new users.

**Comment by stevenbell** [comment]
Using `dconf` it's possible to load a gTile configuration from a text file (see #121), and it's not too hard to add hooks for monitor plug/unplug events.  I'll probably come back to this in a couple weeks, as I recently got a big external monitor for my laptop and switch back and forth frequently.

**Comment by scherepanov** [comment]
Code in master "probably" has a fix for stale monitor handle.
Can you try to install from master and see if new monitor is being properly dynamically detected.

**Comment by stevenbell** [comment]
Here's my little script to read different configurations depending on whether I've got my 4K external monitor plugged in.

```
#! /bin/bash
CONFDIR=/home/steven/tweaks/gtile_switcher

if xrandr | grep --silent "DP-1 connected .* 3840x2160"; then
  CONF=4k
else
  CONF=2k
fi

dconf load /org/gnome/shell/extensions/gtile/ < $CONFDIR/gtile_settings_$CONF

```
I used GNOME to bind this to F7, so it's pretty easy to run albeit not automatic.
